docs/docs/firmware/dev-setup.md

```diff
@@ -173,3 +173,28 @@ _If you use a different IDE, consider [adding instructions](../tutorials/site-de
         1. Select the `clangd` extension and open :octicons-gear-24: Extension Settings.
         
         1. Paste the full `clangd` executable path in the "Clangd: Path" setting.
+
+
+## Pre-Commit Setup
+
+To ensure that we run formatting and code checks before pushing their code, we use `pre-commit` hooks. This helps catch issues locally before code is committed.
```

Comment:
Todo: fix grammar mistake their -> our

Reply:
You could rephrase this as

"We use `pre-commit` hooks to run formatting and code checks before code is pushed."